### ISSUE TITLE
508-defect-2 [SEMANTIC MARKUP]: Search results should be semantically written consistently 

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#26353]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## User Story or Problem Statement

As a screen reader user, I expect:
- search results to be contained within a list `ul` with a `role="list"`, so I can hear an accurate count of how many results exist and where I am within those results
- titles of search results to be headings, so I can quickly navigate through them using a headings list

## Details

Search results are semantically written in different formats. This inconsistency can break habituation for how screen reader users navigate search results. For example, if on one page a user can search by a headings list (resources) but can't on another (va.gov search results), they may accidentally interpret that no results have rendered (on va.gov search results).

## Acceptance Criteria

- [ ] All search result titles are headings (Angela, who is a life long screen reader, has stated headings in search results are important for her to quickly navigate results)
- [ ] All search results are contained within a list `ul` with `role="list"` (all current `ul`s do not have `role="list"` and find a VA form is in a `dl`)

## Recommendation for Find a VA Form list

Search results are compiled together in one large description `dl` list making it hard to determine where one form ends and another begins on some screen readers. It may be better to split up search results by `ul` (to be consistent with other search results) and `h3` paired with `dl`.

### Before
```html
<dl> <!-- Upon entering description list, will be announced as "description list, 11 items" which mirrors number of definition titles and descriptions -->
  <!-- Search result 1 -->
  <dt> <!-- Upon navigating into list item (title), will be announced as -->
    <p>Form 10-10ez</p>
    <a>Instructions and Enrollment Application for Health Benefits</a>
  </dt>
  <dd> <!-- description details, the description, definition, or value for the preceding term -->
    <dfn>Form last updated:</dfn> <!-- definition element, the term being defined within the context of a definition phrase or sentence -->
      01-17-2020
  </dd>
  <dd> <!-- description details, the description, definition, or value for the preceding term -->
    <dfn>Related to:</dfn> <!-- definition element, the term being defined within the context of a definition phrase or sentence -->
      Health care
  </dd>
  <dd> <!-- link to 10-10ez tool -->
    <a>Go to online tool for 10-10EZ Instructions and Enrollment Application for Health Benefits</a>
  </dd>
  <dd> <!-- download link to 10-10ez form -->
    <a>Download form 10-10EZ (PDF)</a>
  </dd>

    <!-- Search result 2 -->
  <dt> <!-- description title, a term in a description or definition list -->
    <p>Form 10-10ez</p>
    <a>Instructions and Enrollment Application for Health Benefits</a>
  </dt>
  <dd> <!-- description details, the description, definition, or value for the preceding term -->
    <dfn>Form last updated:</dfn> <!-- definition element, the term being defined within the context of a definition phrase or sentence -->
      01-17-2020
  </dd>
  <dd> <!-- description details, the description, definition, or value for the preceding term -->
    <dfn>Related to:</dfn> <!-- definition element, the term being defined within the context of a definition phrase or sentence -->
      Health care
  </dd>
  <dd> <!-- link to 10-10ez tool -->
    <a>Go to online tool for 10-10EZ Instructions and Enrollment Application for Health Benefits</a>
  </dd>
  <dd> <!-- download link to 10-10ez form -->
    <a>Download form 10-10EZ (PDF)</a>
  </dd>
</dl>
```

### Suggested
```html
<ul role="list"> <!-- Upon entering list, will be announced as "List, 3 items" which mirrors number of search results-->

<li> <!-- Upon navigating into list, will be announced as 1 of [Total number of results] -->
  <p id="unique-id-1">Form 10-10EZ</p> <!-- unique id ties this to the `h3` which screen reader users might jump directly to via the headings list -->
  <h3 aria-describedby="unique-id-1"> <!-- h3 sets clear expectation that this is the start of a new result, also makes it discoverable in headings list -->
    <a>
      Instructions and Enrollment Application for Health Benefits
    </a>
  </h3>
  <dl>
    <dt>Form last updated:</dt>
    <dd>01-17-2020</dd>
    <dt>Related to:</dt>
    <dd>Healthcare</dd>
  </dl>
  <a>Go to online tool</a> <!-- not technically describing the form so much as it is just a link -->
  <a>Download form 10-10EZ (PDF)</a> <!-- not technically describing the form so much as it is just a link -->
</li>

<li> <!-- Upon navigating into list item, will be announced as 2 of [Total number of results] -->
  <p id="unique-id-2">Form 10-10EZ</p> <!-- unique id ties this to the `h3` which screen reader users might jump directly to via the headings list -->
  <h3 aria-describedby="unique-id-2"> <!-- h3 sets clear expectation that this is the start of a new result, also makes it discoverable in headings list -->
    <a>
      Instructions and Enrollment Application for Health Benefits
    </a>
  </h3>
  <dl>
    <dt>Form last updated:</dt>
    <dd>01-17-2020</dd>
    <dt>Related to:</dt>
    <dd>Healthcare</dd>
  </dl>
  <a>Go to online tool</a> <!-- not technically describing the form so much as it is just a link -->
  <a>Download form 10-10EZ (PDF)</a> <!-- not technically describing the form so much as it is just a link -->
</li>

</ul>
```

**Comment by denisecoveyduc** [comment]
Hi @MarciMcGuireGCIO  This is UI Consistency feedback we have received for Find a Form. Passing it over to your team but please let me know if we can help in any way.

**Comment by joshkimux** [comment]
@zacharymorel let's work on this one together if it's feasible! 🥳

**Comment by zacharymorel** [comment]
Sounds good @joshkimux

**Comment by zacharymorel** [comment]
@joshkimux 

I'm working on this bug now and I have applied the ID's to the **p** and the **h3** tags. It Voice Over reads the Header first before the P tag. That appears to be an issue since I believe **Form number** should be read before the **Form Title**. 
<img width="807" alt="Screen Shot 2021-08-04 at 1 07 41 PM" src="https://user-images.githubusercontent.com/26075258/128224073-199868bc-0f62-4bfe-9299-8072a9a46d4e.png">

Here are some video recordings of my local changes.

https://user-images.githubusercontent.com/26075258/128224594-c70d215a-c3ba-46b5-9427-5dfd9ffe832d.mov


https://user-images.githubusercontent.com/26075258/128224607-078dde8c-04b7-4882-be7e-330936333c15.mov

**Comment by joshkimux** [comment]
@zacharymorel Nothing to worry about here, this is the expected behavior! `aria-describedby` appends the form title as a description to the heading which matches the way in which it is visually designed (heading is significantly larger and takes immediate visual focus, then folks may see the form number above it).

**Comment by joshkimux** [comment]
<img width="1243" alt="Screen Shot 2021-08-05 at 10 09 50 AM" src="https://user-images.githubusercontent.com/14154792/128364457-eabe37e4-33dc-4c9a-96cf-0cbe34c31c49.png">
Wonderful work @zacharymorel 🥳

**Comment by zacharymorel** [comment]
Thank you @joshkimux !

**Comment by MarciMcGuireGCIO** [comment]
Great work, @zacharymorel !  And thanks, @joshkimux for reviewing!
